### PR TITLE
Set namespace function and e2e test

### PR DESCRIPTION
**Pull request by prachirp**
Users currently have no way to set the namespace of all resources when using Config Sync with unstructured layout.
This function provides a way for users to easily set the namespace field.
Fixes GoogleContainerTools/kpt#683

### CODE REVIEW

**Comment by prachirp** [comment]
cc @mortent @phani I made this function as a workaround for users to set namespaces until we decide on a longer term solution using kpt cfg or within Config Sync. PTAL

**Comment by mortent** [comment]
So this would just add namespace to all resources, regardless of whether they have cluster or namespace scope? That is not ideal, but I don't think it will create any serious problems. Have we verified what the behavior is in that case?

**Comment by prachirp** [comment]
Yes this just adds a namespace to all resources. This may cause tools to give warnings that cluster scoped resources have a namespace specified. In such a case users can use the remove namespace fn, which removes namespace from a hardcoded list of resources which it knows should be cluster scoped (https://github.com/kubeflow/kfctl/tree/master/kustomize-fns/remove-namespace).

File: functions/starlark/set_namespace.star [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# set the namespace on each resource
```
Comment (frankfarzan) [inline functions/starlark/set_namespace.star:1]: Is this documented in the catalog? What prompted this? Also we've discussed adding a toolchain column to the catalog  (Can be follow up after the move).

Comment (prachirp) [reply functions/starlark/set_namespace.star:1]: This is not documented in the catalog, I will add it once it's committed. This was prompted by [this issue](https://github.com/GoogleContainerTools/kpt/issues/683) a user filed noting that there was no easy way to add the namespace field to unstructured configs in order to use them with Config Sync.

The new PR will add a toolchain column to the catalog [here](https://github.com/prachirp/kpt/blob/update-general-docs/site/content/en/guides/consumer/function/catalog/_index.md)

**Pull request merged**